### DISCUSSION TITLE
Parse yaml and export as linux environment variable

### DISCUSSION THREAD

**Discussion by sathishsonofsankar** [mikefarah/yq#2114] [Q&A]
Input yaml file:

    Users:
       - name: user1
         subscriptions:
       - name: user2
         subscriptions:
           - id: netflix12341
             secret: HUJBU7868
           - id: prime31413e
             secret: UYHB7897

My objective is to parse this and export as linux environment variable(user=id of subscription). all with in yq script.

    export user1 = null
    export user2 = netflix12341,prime31413e.
I tried using to_entries and ireduce to export every map entry,But couldn't achieve exporting as environment variable.
@mike.f Any help here will be appreciated.
Thanks in advance.

**Comment by mikefarah** [comment]
You can use the `-output-format=shell/-os` to output as a shell script to set variables:

```
/yq '.Users[] | {.name: ([.subscriptions[].id] | join(","))}' examples/data1.yaml -os
user1=
user2='netflix12341,prime31413e'
```

**Reply by sathishsonofsankar** [reply]
Thanks a ton Mike.That works like a charm.

**Reply by sathishsonofsankar** [reply]
i am also trying to get the output in the below format:
user2 = "netflix12341=HUJBU7868,prime31413e=UYHB7897"

I tired something like this...But map join dint work for me
.name = [.subscriptions[].id]=[.subscriptions[].secret] | join(",").
Any help here will be appreciated.Thanks in advance.

**Reply by mikefarah** [reply]
Tricky:

```
./yq  -os '.Users[] | {.name: ( .subscriptions[] as $i ireduce([]; . + $i.id + "=" + $i.secret) | join(",")  )}' data1.yaml
```

I used `ireduce` to reduce the list of subscription entries into an array of "id=secret" strings, then joined those with a comma

```
user1=
user2='netflix12341=HUJBU7868,prime31413e=UYHB7897'

```

**Reply by mikefarah** [reply]
Bonus - filter out user1:

```
./yq  -os '.Users[] | select(.subscriptions | length > 0) | {.name: ( .subscriptions[] as $i ireduce([]; . + $i.id + "=" + $i.secret) | join(",")  )}' data1.yaml
```
